test/suite/test_rollback_to_stable10.py

```diff
@@ -0,0 +1,183 @@
+        self.large_updates(uri_1, value_a, ds_1, nrows, 50)
+
+        self.large_updates(uri_2, value_d, ds_2, nrows, 20)
+        self.large_updates(uri_2, value_c, ds_2, nrows, 30)
+        self.large_updates(uri_2, value_b, ds_2, nrows, 40)
+        self.large_updates(uri_2, value_a, ds_2, nrows, 50)
+
+        # Verify data is visible and correct.
+        self.check(value_d, uri_1, nrows, 20)
+        self.check(value_c, uri_1, nrows, 30)
+        self.check(value_b, uri_1, nrows, 40)
+        self.check(value_a, uri_1, nrows, 50)
+
+        self.check(value_d, uri_2, nrows, 20)
+        self.check(value_c, uri_2, nrows, 30)
+        self.check(value_b, uri_2, nrows, 40)
+        self.check(value_a, uri_2, nrows, 50)
+
+        # Pin stable to timestamp 60 if prepare otherwise 50.
+        if self.prepare:
+            self.conn.set_timestamp('stable_timestamp=' + timestamp_str(60))
+        else:
+            self.conn.set_timestamp('stable_timestamp=' + timestamp_str(50))
+
+        # Create a checkpoint thread
+        done = threading.Event()
+        ckpt = checkpoint_thread(self.conn, done)
+        ckpt.start()
+
+        # Perform several updates in parallel with checkpoint.
+        self.large_updates(uri_1, value_e, ds_1, nrows, 70)
+        self.large_updates(uri_2, value_e, ds_2, nrows, 70)
+        self.large_updates(uri_1, value_f, ds_1, nrows, 80)
+        self.large_updates(uri_2, value_f, ds_2, nrows, 80)
+
+        done.set()
+        ckpt.join()
+
+        # Simulate a server crash and restart.
+        self.simulate_crash_restart(".", "RESTART")
+
+        # Check that the correct data is seen at and after the stable timestamp.
+        self.check(value_a, uri_1, nrows, 50)
+        self.check(value_a, uri_1, nrows, 80)
+        self.check(value_b, uri_1, nrows, 40)
+        self.check(value_c, uri_1, nrows, 30)
+        self.check(value_d, uri_1, nrows, 20)
+
+        # Check that the correct data is seen at and after the stable timestamp.
+        self.check(value_a, uri_2, nrows, 50)
+        self.check(value_a, uri_2, nrows, 80)
+        self.check(value_b, uri_2, nrows, 40)
+        self.check(value_c, uri_2, nrows, 30)
+        self.check(value_d, uri_2, nrows, 20)
+
+        stat_cursor = self.session.open_cursor('statistics:', None, None)
+        calls = stat_cursor[stat.conn.txn_rts][2]
+        hs_removed = stat_cursor[stat.conn.txn_rts_hs_removed][2]
+        hs_sweep = stat_cursor[stat.conn.txn_rts_sweep_hs_keys][2]
+        keys_removed = stat_cursor[stat.conn.txn_rts_keys_removed][2]
+        keys_restored = stat_cursor[stat.conn.txn_rts_keys_restored][2]
+        pages_visited = stat_cursor[stat.conn.txn_rts_pages_visited][2]
+        upd_aborted = stat_cursor[stat.conn.txn_rts_upd_aborted][2]
+        stat_cursor.close()
+
+        self.assertEqual(calls, 0)
+        self.assertEqual(keys_removed, 0)
+        self.assertEqual(keys_restored, 0)
+        self.assertGreaterEqual(upd_aborted, 0)
```

Comment:
I think this stat should be greater than zero, to ensure we can execute the scenario we anticipate. is my understanding wrong ?

Reply:
Thanks. I updated the test to ensure that the test verifies the history sweep should be greater than zero.